Reorganise episodes (#26)
ImperialCollegeLondon/intermediate_grad_school_git_course: _episodes/l3-03-colaborative-development.md

> ---

Reviewer: Does it make more sense for this filename to be `l2-03...` rather than `l3-03...` now the course is being split into two sessions?

title: "Collaborative development"
teaching: 5
exercises: 25
questions:
- "How do I put into practice all the previous knowledge at once?"
- "What caveats might I find in a real collaborative scenario?"
objectives:
- Create a pull request from a branch within a repository.
- Create a pull request from a forked repository.
- Manage other people's contributions.
- Create releases at key points of the development.
keypoints:

> - Working collaboratively require coordination - use Issues to discuss with your colleagues who is doing what.

Reviewer: ```suggestion
- Working collaboratively requires coordination - use Issues to discuss with your colleagues who is doing what.
```

> ## Enabling issues in forks
>
> By default, when you fork a repository, Issues are disabled. To enable
> them go to `Settings` in the upper right corner, then to `Options` in the
> left panel and, finally, scroll down to the `Features` section. There
> click the Issues tickbox to enable them.
{: .callout}

> ## Making a book of recipes
>
> Together with some colleagues, you are writing a book of recipes for sauces
> and you are using git for version control and GitHub to collaborate in the
> writing of the book.
>
> Form groups of 3-4 people and choose one to act as administrator. This
> person should:
>- Fork the [Book of Recipes repository](https://github.com/ImperialCollegeLondon/book_of_recipes)
> with the skeleton of the book.
>- Pass the link to the repo to your colleagues.
>
> Now, start collaborating!
> - The administrator creates a new release, let's say `1.0.0`, as the starting point of
>   the book.
> - All, including the administrator, open new issues with recipes for sauces you
>  will like to have in the book.
> - Administrator, add some tags, prioritising some of the recipes, and assign
>  yourself or one of your colleagues as responsible for each of them.

> >  Remember you will need to "Claim the Issue" first in order to be assigned

Reviewer: I don't think this is explicitly "discussed above" now? May need to re-explain e.g. 

> If you do not have write access to the repository (which is the case here) and you are not part of the same organisation of the repository, the only way of being assigned to an Issue is by making a comment on the Issue.

>  to it, as discussed above.
> - Fork the administrator's repository. Administrator, did you notice how the
>  number of forks increases? Which GitHub users forked it from you? And from
>  the original repo?
> - Work on the recipes you have been assigned. Practice the concepts learnt
>  in previous episodes about cloning a repository, making the changes locally
>  and pushing those changes back to the remote repository. You can even try a
>  [gitflow aproach](https://nvie.com/posts/a-successful-git-branching-model/)
>  if you feel ambitious!

> > - When ready, open a PR to the administrator's repo and request his/her review.

Reviewer: ```suggestion
> - When ready, open a PR to the administrator's repo and request their review.
```
